stacker/actions/base.py

```diff
@@ -98,12 +95,21 @@ def s3_stack_push(self, blueprint, force=False):
         """
         key_name = stack_template_key_name(blueprint)
         template_url = self.stack_template_url(blueprint)
-        if self.cfn_bucket.get_key(key_name) and not force:
+        self.ensure_cfn_bucket()
+        try:
+            template_exists = self.s3_conn.head_object(Bucket=self.bucket_name, Key=key_name) is not None
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                template_exists = False
+            else:
+                raise
+
+        if template_exists and not force:
             logger.debug("Cloudformation template %s already exists.",
                          template_url)
             return template_url
-        key = self.cfn_bucket.new_key(key_name)
-        key.set_contents_from_string(blueprint.rendered, encrypt_key=True)
+        self.s3_conn.put_object(Bucket=self.bucket_name, Key=key_name,
+                                      Body=blueprint.rendered, ServerSideEncryption='AES256')
```

Comment:
I don't think this will pass pep8 - in general, can you also make lines < 80.  We've kept the code base at that for the most part, despite the limit being raised to 110.